rotkehlchen/tests/unit/test_data_updates.py

```diff
@@ -553,3 +599,64 @@ def test_accounting_rules_updates(data_updater: RotkiDataUpdater) -> None:
+    with globaldb.conn.read_ctx() as cursor:
+        assert cursor.execute('SELECT COUNT(*) FROM location_asset_mappings').fetchone()[0] == 1500
+
+        for addition in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['additions']:
+            assert cursor.execute(  # additions are not present already
+                'SELECT COUNT(*) FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?', (  # noqa: E501
+                    None if addition['location'] is None else
+                    Location.deserialize(addition['location']).serialize_for_db(),
+                    addition['location_symbol'],
+                ),
+            ).fetchone()[0] == 0
+
+        for update in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['updates']:
+            assert cursor.execute(  # mappings to be updated are not updated already
+                'SELECT local_id FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?',  # noqa: E501
+                (Location.deserialize(update['location']).serialize_for_db(), update['location_symbol']),  # noqa: E501
+            ).fetchone()[0] != update['asset']
+
+        for deletion in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['deletions']:
+            assert cursor.execute(  # mappings to be deleted are present already
+                'SELECT COUNT(*) FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?',  # noqa: E501
+                (Location.deserialize(deletion['location']).serialize_for_db(), deletion['location_symbol']),  # noqa: E501
+            ).fetchone()[0] == 1
+
+    with patch(
+        'requests.get',
+        wraps=make_single_mock_github_data_response(UpdateType.LOCATION_ASSET_MAPPINGS),
+    ):
+        data_updater.check_for_updates()
+
+    with globaldb.conn.read_ctx() as cursor:
+        assert cursor.execute('SELECT COUNT(*) FROM location_asset_mappings').fetchone()[0] == 1500
+
+        for addition in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['additions']:
+            assert cursor.execute(  # additions are added now
+                'SELECT local_id FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?', (  # noqa: E501
+                    None if addition['location'] is None else
+                    Location.deserialize(addition['location']).serialize_for_db(),
+                    addition['location_symbol'],
+                ),
+            ).fetchone()[0] == addition['asset']
+
+        for update in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['updates']:
+            assert cursor.execute(  # updates have been made
+                'SELECT local_id FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?',  # noqa: E501
+                (Location.deserialize(update['location']).serialize_for_db(), update['location_symbol']),  # noqa: E501
+            ).fetchone()[0] == update['asset']
```

Comment:
can't you create an auxiliar function to avoid repeating the code in those checks? It could take an argument `check_before_update: bool`

and then you can do something like

```suggestion
        for update in LOCATION_ASSET_MAPPINGS_DATA['location_asset_mappings']['updates']:
            local_id = cursor.execute(  # updates have been made
                'SELECT local_id FROM location_asset_mappings WHERE location IS ? AND exchange_symbol IS ?',  # noqa: E501
                (Location.deserialize(update['location']).serialize_for_db(), update['location_symbol']),  # noqa: E501
            ).fetchone()[0]
            
            if check_before_update is True:
                 assert  == update['asset']
            else:
                assert != update['asset'] 
```

same for the delete check, you can do both in the same function and avoid repeating code.

Reply:
I believe you can use XOR to do it in one assert line